talk/owt/sdk/p2p/p2ppeerconnectionchannel.cc

```diff
@@ -151,13 +151,32 @@ rtc::scoped_refptr<webrtc::PeerConnectionInterface> P2PPeerConnectionChannel::Ge
   return temp_pc_;
 }
 
+void P2PPeerConnectionChannel::ClearPendingStreams() {
+  RTC_LOG(LS_INFO) << "Clearing pending streams.";
+  {
+    std::lock_guard<std::mutex> lock(pending_publish_streams_mutex_);
+    pending_publish_streams_.clear();
+  }
+  {
+    std::lock_guard<std::mutex> lock(pending_unpublish_streams_mutex_);
+    pending_unpublish_streams_.clear();
+  }
+}
+
 void P2PPeerConnectionChannel::Publish(
     std::shared_ptr<LocalStream> stream,
     std::function<void()> on_success,
     std::function<void(std::unique_ptr<Exception>)> on_failure) {
   RTC_LOG(LS_INFO) << "Publishing a local stream.";
   // Add reference to peer connection until end of function.
   rtc::scoped_refptr<webrtc::PeerConnectionInterface> temp_pc_ = GetPeerConnectionRef();
+  if (!temp_pc_) {
+    RTC_LOG(LS_INFO) << "Peer connection closed, returning.";
+    // Skip on_success callback because we haven't published anything.
```

Comment:
actually good point on calling this out, I missed this on the first read. we should call `on_success` or `on_failure` in all cases. In this case, `on_failure`. Otherwise, I suspect this will lead to a failed audit report and mess up the rate limiter statistics.

See here: https://github.com/AmbientAI/appliance/blob/fa53a7db09a7af7225e903c239d937e76a3659b9/streaming/webrtc_server/src/ambientclient.cc#L1100